include/deal.II/lac/vector.templates.h

```diff
@@ -796,11 +791,122 @@ Vector<Number>::Vector (const TrilinosWrappers::Vector &v)
 
 
 template <typename Number>
+inline
+Vector<Number> &
+Vector<Number>::operator= (const Vector<Number> &v)
+{
+  if (PointerComparison::equal(this, &v))
+    return *this;
+
+  if (vec_size != v.vec_size)
+    reinit (v, true);
+
+  dealii::internal::Vector_copy<Number,Number> copier;
+  copier.dst = val;
+  copier.src = v.val;
+  internal::vectorized_transform(copier,vec_size,thread_loop_partitioner.get());
+
+  return *this;
+}
+
+
+
+#ifdef DEAL_II_WITH_CXX11
+template <typename Number>
+inline
+Vector<Number> &
+Vector<Number>::operator= (Vector<Number> &&v)
+{
+  Subscriptor::operator=(std::move(v));
+
+  if (val) deallocate();
+
+  vec_size = v.vec_size;
+  max_vec_size = v.max_vec_size;
+  val = v.val;
+  thread_loop_partitioner = std::move(v.thread_loop_partitioner);
+
+  v.vec_size = 0;
+  v.max_vec_size = 0;
+  v.val = nullptr;
+
+  return *this;
+}
+#endif
+
+
+
+template <typename Number>
+template <typename Number2>
+inline
+Vector<Number> &
+Vector<Number>::operator= (const Vector<Number2> &v)
+{
+  if (vec_size != v.vec_size)
+    reinit (v, true);
+
+  dealii::internal::Vector_copy<Number,Number2> copier;
+  copier.dst = val;
+  copier.src = v.val;
+  internal::vectorized_transform(copier,vec_size,thread_loop_partitioner.get());
+
+  return *this;
+}
+
+
+
+template <typename Number>
+inline
+void Vector<Number>::reinit (const size_type n,
+                             const bool omit_zeroing_entries)
+{
+  thread_loop_partitioner.reset();
+  if (n==0)
+    {
+      if (val) deallocate();
+      val = 0;
+      max_vec_size = vec_size = 0;
+      return;
+    };
+
+  if (n>max_vec_size)
+    {
+      if (val) deallocate();
+      max_vec_size = n;
+      allocate();
+    };
+  vec_size = n;
+  thread_loop_partitioner.reset(new parallel::internal::TBBPartitioner());
+  if (omit_zeroing_entries == false)
+    *this = static_cast<Number>(0);
+}
```

Comment:
Yes, I think it would be good to add a check. If someone uses a ```Vector``` in a loop over cells with ```hp::DoFHandler``` you may need to do a lot of ```reinit(n_dofs_per_cell)```. Of course, in that case multithreading is probably useless...